### PR TITLE
[frontend] Cart Section

### LINKED ISSUE #13: Port Figma into frontend
just do the skeleton screen first.
implement the interface first then implement the functionality.

## Check how we're doing
We will pr when a section is ready.

✔️: done and merged
⌛: pr and waiting for review
❌: not finished

### (A) Home & Error ✔️
- [x] / (index)
- [x] (401)
- [x] (402)
- [x] (403)

### (B) Discover & News ❌
- [x] /news/newsID
- [ ] /discover?pageNumber={pageNumber}

### (C) Login & Signup ✔️
- [x] /login
- [x] /signup

### (D) Info Settings ✔️
- [x] /user/info
- [x] /seller/info

### (E) ViewShop ✔️
- [x] /sellerID/shop
- [x] /seller/manageProducts
- [x] /seller/manageProducts/new
- [x] /seller/manageProducts/productID

### (F) Order ✔️
- [x] /buyer/order
- [x] /buyer/order/orderID
- [x] /seller/order/orderID
- [x] /seller/order

### (G) Cart ⌛
include cart view and the detail part
- [x] /buyer/cart 

### (H) Product ❌
- [ ] /product

###  (I) Coupon ✔️
- [x] /admin/manageCoupon
- [x] /admin/manageCoupon/couponID
- [x] /admin/manageCoupon/new
- [x] /seller/manageCoupon
- [x] /seller/manageCoupon/new
- [x] /seller/manageCoupon/couponID
- [x] /sellerCoupon
- [x] /coupon

### (J) Admin ManageUser ✔️
- [x] /admin/manageUser

### (K) Security ✔️
- [x] /user/security
- [x] /user/security/password
- [x] /user/security/manageCreditCard
- [x] /user/security/manageCreditCard/new

### (L) Search ❌
include search not found
- [ ] /search
- [ ] /search?shop={keyword}
- [ ] /search?keyword={keyword}

### (M) Report ❌
- [ ] /seller/report
- [ ] /seller/report/new
- [ ] /seller/report/reportID
- [ ] (the missing admin report)

### PR DESCRIPTION
**Pull request by UGisBusy**
Reeeeesolves #13 partially

### (G) Cart 
- [x] [/buyer/cart](http://localhost:5173/buyer/cart) 

Navigate: 
- click the cart icon on nav bar
- use [url](http://localhost:5173/buyer/cart) 

Function:
- most butions will print something to console. will change to use api later.
- click ``checkout`` button in the cart**s** view will bring up ``chekcout`` section of this cart.
- click ``+`` by the discount will bring up ``usable coupon``  selection.

### Testing Approach (if unconventional)

- open frontend server
  ```sh
  cd frontend
  npm run dev
  ```
- go the the new page

### Additional context

- The checkout section will show one of the same cart because there're only one mocked data.
- Buttons change nothing in ui because it only refetches data and the mocked data cant be modify obviously.
- Credit card selection on checkout page is wip because the api doc man fogor.
- No idea how to handle status code other ok in ``useQuery``. Currently i just link to error pages. Pls give advice if u have any ❤️

### CODE REVIEW

**Review by KatoToyoki — CHANGES_REQUESTED** [review]
image
image

this part in checkout, the discount reveals the chosen ones while the modal reveals the available but unchosen ones could be more intuitive for users.

File: frontend/src/components/CouponItemTemplate.tsx [hunk 1]
```diff
@@ -21,6 +22,7 @@ const couponStyle = {
   borderRadius: '30px',
   padding: '5%',
   border: 'var(--border) solid 2px',
+  witdh: '100%',
```
Comment (NOOBDY) [inline frontend/src/components/CouponItemTemplate.tsx]:
```suggestion
  width: '100%',
```

File: frontend/src/components/Cart.tsx [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+  products: ProductProps[];
+}
+
+interface CouponProps {
+  description: string;
+  discount: number;
+  id: number;
+  name: string;
+  type: string; // 'percentage' | 'fixed' | 'shipping'
+  scope: string; // 'global' | 'shop'
```
Comment (NOOBDY) [inline frontend/src/components/Cart.tsx]:
```suggestion
  type: 'percentage' | 'fixed' | 'shipping';
  scope: 'global' | 'shop';
```

File: frontend/src/pages/user/buyer/cart/index.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import CartGroup from '@components/Cart';
```
Comment (NOOBDY) [inline frontend/src/pages/user/buyer/cart/index.tsx]:
```suggestion
import Cart from '@components/Cart';
```

File: frontend/src/pages/user/buyer/history/[historyID].tsx [hunk 4]
```diff
@@ -2,13 +2,13 @@ import { faFile, faMoneyBill, faTruck, faBox } from '@fortawesome/free-solid-svg
 import { Col, Row } from 'react-bootstrap';
 import { useParams } from 'react-router-dom';
 
-import CartItem from '@components/CartItem';
+import CartItem from '@components/CartProduct';
```
Comment (NOOBDY) [inline frontend/src/pages/user/buyer/history/[historyID].tsx]:
```suggestion
import CartProduct from '@components/CartProduct';
```

**Review by NOOBDY — CHANGES_REQUESTED** [review]

File: frontend/src/components/Cart.tsx [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+import { faCircleXmark } from '@fortawesome/free-regular-svg-icons';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+
+import CartProduct from '@components/CartProduct';
+import sellerInfo from '@pages/user/seller/sellerInfo.json';
+import UserItem from '@components/UserItem';
+import TButton from './TButton';
+import CouponItemTemplate from './CouponItemTemplate';
+import CheckoutItem from './CheckoutItem';
+import CheckoutItemCoupon from './CheckoutItemCoupon';
```
Comment (KatoToyoki) [inline frontend/src/components/Cart.tsx]: here should use `@components/` rather than `./`, otherwise lgtm

**Review by KatoToyoki — CHANGES_REQUESTED** [review]

**Review by KatoToyoki — DISMISSED** [review]

**Review by KatoToyoki — DISMISSED** [review]

**Review by NOOBDY — APPROVED** [review]

**Review by KatoToyoki — APPROVED** [review]

**Pull request merged**